rust/parquet/src/arrow/arrow_writer.rs

```diff
@@ -430,6 +472,28 @@ fn get_primitive_def_levels(
     primitive_def_levels
 }
 
+macro_rules! def_get_binary_array_fn {
+    ($name:ident, $ty:ty) => {
+        fn $name(array: &$ty) -> Vec<ByteArray> {
+            let mut values = Vec::with_capacity(array.len() - array.null_count());
+            for i in 0..array.len() {
+                if array.is_valid(i) {
+                    let buffer = BufferPtr::new(array.value(i).as_bytes().into());
```

Comment:
This looks longer than `ByteArray::from(array.value(i).to_vec())`. Given that the difference btwn BinaryArray and StringArray is that the latter has valid utf8 bytes, when you only want the bytes, it's convenient to reinterpret StringArray as BinaryArray using `BinaryArray::from(array.data())`.

Reply:
:+1: